modules/network/src/main/java/org/apache/ignite/internal/network/netty/ChannelCreationListener.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.netty;
+
+import java.util.concurrent.CompletableFuture;
+
+/** Channel creation listener. */
+public interface ChannelCreationListener {
+    /**
+     * Notifies of creation of a new channel before the handshake.
+     *
+     * @param consistentId Remote node's consistent id.
+     * @param channelId Channel id.
+     * @return A future which is completed when listener's action is done.
+     */
+    CompletableFuture<Void> notifyInboundChannelCreation(String consistentId, short channelId);
```

Comment:
I don't see a unique launch ID. Is it not available yet?
If it's not available, then how do you know node's consistent ID?
Or maybe it's not required here. You see, it's hard to understand your intentions with this interface.

Reply:
Removed the method altogether